### PR TITLE
Update notebook examples

### PR DESCRIPTION
**Pull request by NvTimLiu**
1, Add variables to overwrite the spark resource configs, from https://github.com/NVIDIA/spark-rapids-examples/pull/49

2, Use the consistent `DATA_ROOT` environment to overwrite the root path of the datase, refer to https://github.com/NVIDIA/spark-rapids-examples/pull/34

Signed-off-by: Tim Liu <timl@nvidia.com>

### CODE REVIEW

File: examples/Spark-ETL+XGBoost/taxi/notebooks/python/taxi-ETL.ipynb [hunk 1]
```diff
@@ -85,9 +85,9 @@
     "# from https://www1.nyc.gov/site/tlc/about/tlc-trip-record-data.page\n",
     "# or you can just unzip datasets/taxi-small.tar.gz and use the provided\n",
     "# sample dataset datasets/taxi/taxi-etl-input-small.csv\n",
-    "dataRoot = os.getenv('DATA_ROOT', '/root')\n",
-    "rawPath = dataRoot + '/spark-rapids-examples/datasets/taxi/taxi-etl-input-small.csv'\n",
-    "outPath = dataRoot + '/spark-rapids-examples/datasets/taxi/output'"
+    "dataRoot = os.getenv('DATA_ROOT', '/data')\n",
```
Comment (NvTimLiu) [inline examples/Spark-ETL+XGBoost/taxi/notebooks/python/taxi-ETL.ipynb:88]: Use the consistent DATA_ROOT environment , refer to #49

File: examples/Spark-ETL+XGBoost/taxi/notebooks/scala/taxi-ETL.ipynb [hunk 2]
```diff
@@ -120,9 +120,9 @@
     }
    ],
    "source": [
-    "val dataRoot = sys.env.getOrElse(\"DATA_ROOT\", \"/root\")\n",
-    "val rawPath = dataRoot + \"/spark-rapids-examples/datasets/taxi/taxi-etl-input-small.csv\"\n",
-    "val outPath = dataRoot + \"/spark-rapids-examples/datasets/taxi/output\""
+    "val dataRoot = sys.env.getOrElse(\"DATA_ROOT\", \"/data\")\n",
```
Comment (NvTimLiu) [inline examples/Spark-ETL+XGBoost/taxi/notebooks/scala/taxi-ETL.ipynb:123]: Use the consistent DATA_ROOT environment , refer to #34

File: examples/micro-benchmarks/notebooks/micro-benchmarks-cpu.ipynb [hunk 3]
```diff
@@ -63,19 +63,23 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "# You need to update data path to your real path!\n",
+    "# You need to update data path with your real path and hardare resource!\n",
```
Comment (NvTimLiu) [inline examples/micro-benchmarks/notebooks/micro-benchmarks-cpu.ipynb:66]: Add variables to overwrite the spark resource configs, from #49;

We need overwrite the resources configurations on specified environments , e.g. CI hosts may not have so many CPU/GPU/Mem resources.

**Comment by NvTimLiu** [comment]
This PR should also be cherry-pick to branch-21.12

**Comment by nvliyuan** [comment]
we have three additional notebooks, should we also add these to ci jobs?
scala/mortgage-ETL.ipynb 
python/taxi-ETL.ipynb
scala/taxi-ETL.ipynb

File: examples/micro-benchmarks/notebooks/micro-benchmarks-cpu.ipynb (hunk 3 above)
Comment (firestarman) [inline examples/micro-benchmarks/notebooks/micro-benchmarks-cpu.ipynb:66]: ?
```suggestion
    "# You need to update data path with your real path and hardware resource!\n",
```

File: examples/micro-benchmarks/notebooks/micro-benchmarks-gpu.ipynb [hunk 4]
```diff
@@ -90,9 +97,9 @@
     "# 4 tasks will run concurrently per GPU\n",
     "conf.set(\"spark.rapids.sql.concurrentGpuTasks\", \"4\")\n",
```
Comment (nvliyuan) [inline examples/micro-benchmarks/notebooks/micro-benchmarks-gpu.ipynb:98]: we may have oom issue while running benchmarks on some week performance GPU like T4 with high concurrentGpuTasks, can we also make concurrentGpuTasks variable?

**Review by nvliyuan — CHANGES_REQUESTED** [review]

**Review by nvliyuan — APPROVED** [review]

**Pull request merged**